cmd/entrypoint/webhooks.go

```diff
@@ -0,0 +1,312 @@
+				k8sTypesCoreV1.TLSCertKey:       caPEM.Bytes(),
+			},
+		}, k8sTypesMetaV1.CreateOptions{})
+		if err != nil {
+			if k8sErrors.IsAlreadyExists(err) {
+				caSecret, err = secrets.Get(ctx, caSecretName, k8sTypesMetaV1.GetOptions{})
+				if err != nil {
+					return err
+				}
+			} else {
+				return err
+			}
+		}
+	}
+
+	// CA Secret already exists, so load data from it
+	if caPEM == nil || caTemplate == nil || caPrivKey == nil {
+		if caSecret == nil {
+			return fmt.Errorf("couldnt get or generate CA secret")
+		}
+
+		// Parse CA Key
+		caPrivKeyPEMBytes, ok := caSecret.Data[k8sTypesCoreV1.TLSPrivateKeyKey]
+		if ok {
+			keyBlock, _ := pem.Decode(caPrivKeyPEMBytes)
+			if key, err := x509.ParsePKCS1PrivateKey(keyBlock.Bytes); err == nil {
+				caPrivKey = key
+			} else {
+				return fmt.Errorf("bad key loaded in CA secret: %s", err.Error())
+			}
+		} else {
+			return fmt.Errorf("no key found in CA secret!")
+		}
+
+		// parse ca cert
+		caPEMBytes, ok := caSecret.Data[k8sTypesCoreV1.TLSCertKey]
+		if ok {
+			// we need the PEM and the parsed cert
+			caPEM = b.NewBuffer(caPEMBytes)
+			certBlock, rest := pem.Decode(caPEMBytes)
+			if string(rest) != "" || certBlock.Type != "CERTIFICATE" {
+				return fmt.Errorf("bad cert loaded in CA secret")
+			}
+			caTemplate, err = x509.ParseCertificate(certBlock.Bytes)
+			if err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("no cert found in CA secret!")
+		}
+	}
+
+	// these need to be pointer vals for some reason
+	webhookConfig := k8sApiExtTypes.CustomResourceConversion{
+		Strategy: k8sApiExtTypes.WebhookConverter,
+		WebhookClientConfig: &k8sApiExtTypes.WebhookClientConfig{
+			Service: &k8sApiExtTypes.ServiceReference{
+				Namespace: GetAmbassadorNamespace(),
+				Name:      GetAdminService(),
+				Path:      constStringPtr(webhookPath),
+				Port:      int32Ptr(webhookPort),
+			},
+			CABundle: caPEM.Bytes(),
+		},
+	}
+
+	//	// get list of CRDs
+	//	types := scheme.KnownTypes(schema.GroupVersion{
+	//		Group: "getambassador.io",
+	//		Version: "v2",
+	//	})
+	apiExtClient, err := k8sApiExtClient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+	crdInterface := apiExtClient.CustomResourceDefinitions()
+	crds, err := crdInterface.List(ctx, k8sTypesMetaV1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	var count int
+	var errs derror.MultiError
+	for _, crd := range crds.Items {
+		if len(crd.Spec.Versions) < 1 || !scheme.Recognizes(schema.GroupVersionKind{
+			Group: crd.Spec.Group,
+			// Versions is a mandatory field we can rely on
+			// to have at least 1 value. Regardless, we
+			// protect against len=0 in the conditional
+			Version: crd.Spec.Versions[0].Name,
+			Kind:    crd.Spec.Names.Kind,
+		}) {
+			continue
+		}
+
+		count += 1
+		crd.Spec.Conversion = &webhookConfig
+		_, err := crdInterface.Update(ctx, &crd, k8sTypesMetaV1.UpdateOptions{})
+		if err != nil {
+			errs = append(errs, err)
+		}
```

Comment:
1. This should avoid calling update if the webhook config already matches.
2. This should ignore `k8sErrors.IsConflict(err)` errors, since it is racing the other replicas.